compaction/compaction_manager.cc

```diff
@@ -1265,6 +1265,13 @@ class offstrategy_compaction_task_executor : public compaction_task_executor {
             try {
                 table_state& t = *_compacting_table;
                 auto maintenance_sstables = t.maintenance_sstable_set().all();
+                auto size = maintenance_sstables->size();
+                if (!size) {
```

Comment:
we added a sstable_set::size() method, let's take this opportunity to kill

auto maintenance_sstables = t.maintenance_sstable_set().all();

and retrieve size directly from set, with  t.maintenance_sstable_set().size();

Reply:
fixed in v2